[tckeong] iP

tckeong opened the pull request:
# JayBot
> Don't be something to everyone, Be everything to someone

JayBot frees your mind of having to remember things you need to do. It's,
- CLI-based
- Easy to learn the command
- Easy to use

All you need to do is,
1. download it from [here](https://github.com/tckeong/ip/releases/tag/A-Jar)
2. Open the folder that the file Jay.jar located
3. using command  ```java -jar Jay.jar```
4. let it manage your tasks for you 

And it is **FREE** !!!

Features:
 - [x] Manage your task
 - [x] Managing deadlines
 - [x] Find task by keyword
---
If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```
public class Main {
    public static void main(String[] args) {
        Jay jay = new Jay("Jay");
        jay.start();
    }
}

```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

tckeong committed:
Level-0

tckeong committed:
Level-1

tckeong committed:
Level-2

tckeong committed:
Level-3

tckeong committed:
Level-4

tckeong committed:
A-TextUiTesting

tckeong committed:
Level-5

tckeong committed:
Level-6

tckeong committed:
Level-6

tckeong committed:
Create h --force --set-upstream origin master

tckeong committed:
A-Enums

tckeong committed:
Add save function

tckeong committed:
Merge branch 'branch-Level-7'

tckeong committed:
Add date and time parse function

tckeong committed:
Merge branch 'branch-Level-8'

tckeong committed:
Refactor the project

tckeong committed:
Create packages. Fixed bug of date and time parsing

tckeong committed:
Create packages. Fixed bug of date and time parsing

tckeong committed:
Merge branch 'master' into add-gradle-support

tckeong committed:
Add gradle support

tckeong committed:
Add JUnit test

tckeong committed:
Add docs

tckeong committed:
Refactor the code to follow coding standard

tckeong committed:
Add find feature

DennieDan commented on src/main/java/Jay.java:
```diff
@@ -0,0 +1,153 @@
+        Command.CommandType commandType = command.getCommandType();
+
+        try {
+            switch (commandType) {
```
Should your switch-case be in the following form?
```Java
switch (condition) {
case ABC:
    statements;
    // Fallthrough
case DEF:
    statements;
    break;
case XYZ:
    statements;
    break;
default:
    statements;
    break;
}
```
You can go to Settings -> Editor -> Code Style -> Java
Choose *Wrapping and Braces* tab and uncheck *Indent 'case' branches* under **'switch' statement/expression**

DennieDan commented on src/main/java/Jay.java:
```diff
@@ -0,0 +1,153 @@
+     * Shows the tasks in the task list.
+     * If the task list is empty, a message will be shown to the user.
+     * @return The tasks in the task list or a message indicating that the task list is empty.
+     */
```
You did this correctly in the previous methods. There should be an empty line between the description of the method and the list of parameters in the JavaDoc.

DennieDan replied:
You did this correctly in the previous methods. There should be an empty line between the description of the method and the list of parameters in the JavaDoc.

DennieDan commented on src/main/java/command/Command.java:
```diff
@@ -0,0 +1,111 @@
+ */
+public class Command {
+    public enum CommandType {
+        List, Add, Mark, Unmark, Delete, Find, Exit
```
Is it better to be consistent in the way you name the enum's values? You are using CamelCase for ``CommandType`` and SCREAMING_SNAKE for ``Task.TYPE``

DennieDan replied:
Is it better to be consistent in the way you name the enum's values? You are using CamelCase for ``CommandType`` and SCREAMING_SNAKE for ``Task.TYPE``

DennieDan commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,133 @@
+    public static Task parseTask(Task.TYPE taskType, Command command) throws InvalidTaskException {
+        try {
+            switch (taskType) {
+                case TODO: {
```
As stated in the Coding Standard, the content of ``case`` shouldn't be inside Egyptian braces, right?

DennieDan replied:
As stated in the Coding Standard, the content of ``case`` shouldn't be inside Egyptian braces, right?

DennieDan commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,133 @@
+     * @param time The LocalTime object to be parsed.
+     * @return The string representation of the LocalTime object.
+     */
+    public static String timeToString(LocalTime time) {
```
This ``timeToString`` method and the names of other 3 methods above seem not to be of a verb type, aren't they?

DennieDan replied:
This ``timeToString`` method and the names of other 3 methods above seem not to be of a verb type, aren't they?

DennieDan commented on src/main/java/task/DeadlineTask.java:
```diff
@@ -0,0 +1,28 @@
+    }
+
+    @Override
+    public String simpleFormat() {
```
``simpleFormat()`` sounds more like a noun than a verb, right?

DennieDan replied:
``simpleFormat()`` sounds more like a noun than a verb, right?

DennieDan commented on src/test/java/TaskListTest.java:
```diff
@@ -0,0 +1,146 @@
+    }
+
+    @Test
+    public void testFindTasks() {
```
>Underscores may be used in test method names using the following three part format ``featureUnderTest_testScenario_expectedBehavior()``
e.g. ``sortList_emptyList_exceptionThrown()`` ``getMember_memberNotFound_nullReturned``
Third part or both second and third parts can be omitted depending on what's covered in the test. For example, the test method ``sortList_emptyList()`` will test ``sortList()`` method for all variations of the 'empty list' scenario and the test method ``sortList()`` will test the ``sortList()`` method for all scenarios.

According to this part of the doc, there is a 3-part format for name of testing methods. However, I'm not sure if you can use other ways as they use the word **may** at the beginning.

DennieDan replied:
>Underscores may be used in test method names using the following three part format ``featureUnderTest_testScenario_expectedBehavior()``
e.g. ``sortList_emptyList_exceptionThrown()`` ``getMember_memberNotFound_nullReturned``
Third part or both second and third parts can be omitted depending on what's covered in the test. For example, the test method ``sortList_emptyList()`` will test ``sortList()`` method for all variations of the 'empty list' scenario and the test method ``sortList()`` will test the ``sortList()`` method for all scenarios.

According to this part of the doc, there is a 3-part format for name of testing methods. However, I'm not sure if you can use other ways as they use the word **may** at the beginning.

DennieDan commented on src/test/java/parser/ParserTest.java:
```diff
@@ -0,0 +1,55 @@
+    }
+
+    @Test
+    public void testParseTime_invalidTime_exceptionThrown() {
```
If you use this 3-part format, I think the first part should be the name of the method i.e. ``parseTime_invalidTime_exceptionThrown``

DennieDan replied:
If you use this 3-part format, I think the first part should be the name of the method i.e. ``parseTime_invalidTime_exceptionThrown``

DennieDan commented on src/main/java/Jay.java:
```diff
@@ -0,0 +1,153 @@
+
+        try {
+            switch (commandType) {
+                case List -> {
```
Should your switch-case be in the following form?
```Java
switch (condition) {
case ABC:
    statements;
    // Fallthrough
case DEF:
    statements;
    break;
case XYZ:
    statements;
    break;
default:
    statements;
    break;
}
```
You can go to Settings -> Editor -> Code Style -> Java
Choose *Wrapping and Braces* tab and uncheck *Indent 'case' branches* under **'switch' statement/expression**

DennieDan reviewed (commented):
So far so good, just some minor mistakes to fix!

tckeong committed:
Add checkstyle

To fulfil the requirement of this IP and also adhere to the java code
styles, add checkstyle plugin to this project. And also refactor the
project.

Add config/checkstyle/checkstyle.xml and
config/checkstyle/suppressions.xml, and also add a global package for
whole project.

tckeong committed:
Add GUI interface

Due to the requirement of this course and make this chatbot easy to
use, added GUI interface.

Add new gui package, collects all the class relevant to javafx gui
interface. Add class Gui, which manage the whole gui, which is
also the stage of javafx. Add class MainWindow, which manage the user
interaction and the response of the chatbot. Add class DialogBox,
easier to manage the dialog box in the user interface. Add resources
directory, manage all the images and the fxml files, make the ui design
become easier.

Collect the classes Gui, MainWindow, DialogBox into one package gui
help to organized the whole project well.

leroytan commented on src/main/java/jay/Jay.java:
```diff
@@ -0,0 +1,174 @@
+ * Jay.Jay is a personal assistant chatbot that helps users to keep track of various tasks.
+ */
+public class Jay {
+    private final String name;
```
Consider changing final variables to CAPS.

leroytan commented on src/main/java/jay/Ui.java:
```diff
@@ -0,0 +1,28 @@
+     * Print the message in a nice format
+     * @param output The message to be printed
+     */
+    public void output(String output) {
```
Consider changing the name to printToScreen

leroytan commented on src/main/java/jay/parser/InvalidTimeException.java:
```diff
@@ -0,0 +1,10 @@
+/**
+ * Represents an exception thrown when the time format is invalid.
+ */
+public class InvalidTimeException extends  Exception {
```
small error, but double spaces

leroytan commented on src/main/java/jay/TaskList.java:
```diff
@@ -0,0 +1,200 @@
+     * @param taskNumber The index of the Jay.task need to be checked.
+     * @throws InvalidCommandException If the Jay.task number is invalid.
+     */
+    private void isValidTaskNumberCheck(int taskNumber) throws InvalidCommandException {
```
consider changing the type to boolean? Or changing the naming so that it sounds less like a boolean?

leroytan reviewed (commented):
LGTM, just some issues with constants and boolean naming

tckeong committed:
Merge branch 'branch-A-Assertions' of github.com:tckeong/ip into branch-A-Assertions

# Conflicts:
#	src/main/java/jay/Jay.java
#	src/main/java/jay/gui/DialogBox.java
#	src/main/java/jay/gui/Gui.java
#	src/main/java/jay/parser/Parser.java

tckeong committed:
Add assertions

To enhance the stability of this application and the requirement of
this course

Adding not null assertions to Jay.addTask function, DialogBox class,
Gui class, Parser.parseTask function, Parser.convertDateToString
function, Parser.convertDateToStorageString function,
Parser.convertTimeToStorageString function and
Parser.convertTimeToString function

tckeong committed:
Improve Code Quality

To improve the readability of the code and also adhere to the coding
standard of the course

Make some changes to the docs, and also refactor the format of switch
case

tckeong committed:
Merge pull request #1 from tckeong/branch-A-Assertions

branch A-Assertions

tckeong committed:
Merge pull request #2 from tckeong/branch-A-CodeQuality

Branch A-CodeQuality

tckeong committed:
Add CI

Due to the course requirement, add CI to this project, and rewrite
the test part of the code

tckeong committed:
Add Priority To Task

To give the user flexibility the adjust the priority of the task

Add the "set" command in Jay chatbot to set the priority of task,
change the format of the Task string, and also the Task storage string
format. Make the MainWindow show the tasks with high priority when the
chatbot start. And also modify the test cases.

tckeong committed:
Add help command

To let the new user to use this chatbot easier, add help command

Add new command "help", return the usage of all available command of
the chatbot when accept the "help" command

tckeong committed:
Improve GUI

tckeong committed:
Make Gui Better

To create a good UX, make some change about gui part

Modify some GUI format, make it is more user friendly. Add a exception
class ResponseMessageException, let the GUI can output different
format if the any exception happened in the bot.

tckeong committed:
Add error handling

To tackle some implicit problem of the task, like the logic error the
start time is after the end time, adding more error handling

Check if the start time is before end time when construct the Event
task

tckeong committed:
Fix empty description bug

Fix the bug that the description part of user input can be empty

Check if the description is empty when parsing the command, and add a
InvalidDescriptionException in task class

tckeong committed:
Add title to the GUI window

tckeong committed:
Add ReadMe

Add ReadMe file of the project, and do some reformat

tckeong committed:
Modify ReadMe

tckeong committed:
Reformat ReadMe

tckeong committed:
Reformat ReadMe

tckeong committed:
Add AI.md

tckeong committed:
Modify ReadMe

tckeong committed:
Modify ReadMe

The table format in github page is wrong, modify the ReadMe, hope can
make it correct

Change the format of ReadMe in /docs

tckeong committed:
Modify ReadMe

The table format in github page is wrong, modify the ReadMe, hope can
make it correct

Modify the format of ReadMe in /docs

tckeong committed:
Modify ReadMe

The table format in github page is wrong, modify the ReadMe, hope can
make it correct

Modify the format of ReadMe in /docs

tckeong committed:
Remove additional file

To make the project more clean

Delete the "h --force --set-upstream origin master" file

tckeong committed:
Fix the format

Fix the format, adhere to CS2103T code style

Make some format change and fix some bugs on the project, especially
the format of the import header
